test/tests/rackhd20/test_rackhd20_api_notifications.py

```diff
@@ -0,0 +1,271 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+Author(s):
+'''
+
+from time import sleep
+import Queue
+import flogging
+import logging
+import pika
+import unittest
+import threading
+import fit_common
+from nose.plugins.attrib import attr
+logs = flogging.get_loggers()
+amqp_queue = Queue.Queue(maxsize=0)
+from pymongo import MongoClient
+from bson.objectid import ObjectId
+
+
+IP = None
+AMQP_PORT = fit_common.fitports()['amqp_ssl']
+MONGO_PORT = fit_common.fitports()['mongo_port']
+HTTP_PORT = fit_common.fitports()['http']
+env_vagrant = True
+
+
+class AmqpWorker(threading.Thread):
+    def __init__(self, exchange_name, topic_routing_key, external_callback, timeout=10):
+        threading.Thread.__init__(self)
+        pika_logger = logging.getLogger('pika')
+        if fit_common.VERBOSITY >= 8:
+            pika_logger.setLevel(logging.DEBUG)
+        elif fit_common.VERBOSITY >= 4:
+            pika_logger.setLevel(logging.WARNING)
+        else:
+            pika_logger.setLevel(logging.ERROR)
+        self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=fit_common.fitargs()["rackhd_host"],
+                                                                            port=AMQP_PORT))
+        self.channel = self.connection.channel()
+        result = self.channel.queue_declare(exclusive=True)
+        queue_name = result.method.queue
+        self.channel.queue_bind(exchange=exchange_name, queue=queue_name, routing_key=topic_routing_key)
+        self.channel.basic_consume(external_callback, queue=queue_name)
+        self.connection.add_timeout(timeout, self.dispose)
+
+    def dispose(self):
+        logs.debug_3('Pika connection timeout')
+        if self.connection.is_closed is False:
+            self.channel.stop_consuming()
+            logs.debug_1("Attempting to close the Pika connection")
+            self.thread_stop = True
+
+    def run(self):
+        logs.debug_7('start consuming')
+        self.channel.start_consuming()
+
+
+@attr(all=True, regression=False, smoke=True)
+class test_alert_notification(unittest.TestCase):
+    CLIENT = None
+    MONGO_PORT = None
+    CATALOGS_COUNT = None
+    NODES_COUNT = None
+    OBMS_COUNT = None
+
+    @classmethod
+    def setUpClass(self):
+        logs.debug_1("MONGO_PORT: {0}, AMQP_PORT: {1}, HTTP_PORT:{2}".format(MONGO_PORT, AMQP_PORT, HTTP_PORT))
+        self.CLIENT = MongoClient('localhost', MONGO_PORT)
+        self.db = self.CLIENT.pxe
+        self.node = {
+            "identifiers": ["FF:FF:FF:FF:FF:FF"],
+            "name": "FF:FF:FF:FF:FF:FF",
+            "relations": [],
+            "tags": [],
+            "type": "compute"
+        }
+        if fit_common.fitargs()['stack'] == 'vagrant_guest':
+            IP = "127.0.0.1"
+        elif fit_common.fitargs()['stack'] == 'vagrant':
+            IP = "10.0.2.2"
+        else:
+            env_vagrant = False # NOQA
```

Comment:
This assignment of env_vagrant won't live outside of setUpClass since env_vagrant is not declared as a global.

Small python example to demostrate:
```
env_vagrant=True

class test_alert_notification():

    @classmethod
    def someClassMethod(self):
        # global env_vagrant
        env_vagrant=False

tan = test_alert_notification()

print env_vagrant
tan.someClassMethod()
print env_vagrant

```